ignite3/src/main/java/site/ycsb/db/ignite3/IgniteSqlClient.java

```diff
@@ -0,0 +1,156 @@
+package site.ycsb.db.ignite3;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.ignite.sql.ResultSet;
+import org.apache.ignite.sql.Session;
+import org.apache.ignite.sql.SqlRow;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import site.ycsb.ByteIterator;
+import site.ycsb.DBException;
+import site.ycsb.Status;
+import site.ycsb.StringByteIterator;
+
+/**
+ * Ignite3 SQL API client.
+ */
+public class IgniteSqlClient extends AbstractSqlClient {
+
+  private static final Logger LOG = LogManager.getLogger(IgniteSqlClient.class);
+
+  protected static Session session;
+
+  private static final AtomicInteger SQL_INIT_COUNT = new AtomicInteger(0);
+
+  @Override
+  public void init() throws DBException {
+    super.init();
+
+    SQL_INIT_COUNT.incrementAndGet();
+
+    synchronized (IgniteSqlClient.class) {
```

Comment:
Shouldn't we use double checked locking (+`volatile`) here?

https://www.baeldung.com/java-singleton-double-checked-locking